core/trino-main/src/main/java/io/trino/server/remotetask/HttpRemoteTask.java

```diff
@@ -166,7 +167,7 @@
 
     private final RequestErrorTracker updateErrorTracker;
 
-    private final AtomicBoolean needsUpdate = new AtomicBoolean(true);
+    private final AtomicInteger needsUpdateVersion = new AtomicInteger();
```

Comment:
The name of this variable is bad. 
I did not come with anything I like very much. But I think even sth like `triggerUpdateCounter` would be better.